src/harness/fourslashImpl.ts

```diff
@@ -822,6 +822,27 @@ export class TestState {
         });
     }
 
+    public baselineInlayHints(span: ts.TextSpan = { start: 0, length: this.activeFile.content.length }, preferences?: ts.UserPreferences): void{
+        interface HasPosition { position: number; }
+        const sortHints = (a: HasPosition, b: HasPosition) => {
+            return a.position - b.position;
+        };
+
+        const baselineFile = this.getBaselineFileNameForContainingTestFile();
+        const fileName = this.activeFile.fileName;
+        const hints = this.languageService.provideInlayHints(fileName, span, preferences);
+        const annotations = ts.map(hints.sort(sortHints), hint => {
+            const span = { start: hint.position, length: hint.text.length };
+            const startLc = this.languageServiceAdapterHost.positionToLineAndCharacter(fileName, span.start);
+            const underline = " ".repeat(startLc.character) + "^";
+            let annotation = this.getFileContent(fileName).split(/\r?\n/)[startLc.line];
+            annotation += "\n" + underline + "\n" + JSON.stringify(hint, undefined, "  ");
+            return annotation;

```

Comment:
Kind of wasteful. Here's how we do it based on reporting diagnostics:

```suggestion
            const sourceText = this.getFileContent(fileName);
            const lastLineIndex = getLineAndCharacterOfPosition(file, file.text.length).line
            const endPosition = startLc.line < lastLineIndex ? getPositionOfLineAndCharacter(file, i + 1, 0) : file.text.length;
            let lineContent = file.text.slice(, lineEnd);
            const annotation = lineContent.trimEnd() + "\n" + underline + "\n" + JSON.stringify(hint, undefined, "  ");
```

Reply:
What is the `SourceFileLike` parameter for `getLineAndCharacterOfPosition` that I would use here?

Reply:
Would `this.activeFile` work?

Reply:
nope

Reply:
or well yeah I can adjust the object for the call

Reply:
Although idk I prefer to leave it as is.